src/script/components/message/ContentMessage/index.tsx

```diff
@@ -0,0 +1,327 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import React from 'react';
+
+import {QualifiedId} from '@wireapp/api-client/src/user';
+import {useKoSubscribableChildren} from 'Util/ComponentUtil';
+
+import {Conversation} from 'src/script/entity/Conversation';
+import {ContentMessage} from 'src/script/entity/message/ContentMessage';
+import {EphemeralStatusType} from '../../../message/EphemeralStatusType';
+
+import {Asset} from 'src/script/entity/message/Asset';
+import type {FileAsset} from 'src/script/entity/message/FileAsset';
+import type {MediumImage} from 'src/script/entity/message/MediumImage';
+import type {Text} from 'src/script/entity/message/Text';
+import type {Location} from 'src/script/entity/message/Location';
+
+import Avatar, {AVATAR_SIZE} from 'Components/Avatar';
+import MessageQuote from './MessageQuote';
+import MessageLike from './MessageLike';
+import MessageFooterLike from './MessageFooterLike';
+import FileAssetComponent from './asset/FileAssetComponent';
+import AudioAssetComponent from './asset/AudioAsset';
+import ImageAssetComponent from './asset/ImageAsset';
+import LinkPreviewAssetComponent from './asset/LinkPreviewAssetComponent';
+import LocationAssetComponent from './asset/LocationAsset';
+import VideoAssetComponent from './asset/VideoAsset';
+import ButtonAssetComponent from './asset/MessageButton';
+import ReadReceiptStatus from '../ReadReceiptStatus';
+import EphemeralTimer from '../EphemeralTimer';
+
+import {AssetType} from '../../../assets/AssetType';
+import {Button} from 'src/script/entity/message/Button';
+import {CompositeMessage} from 'src/script/entity/message/CompositeMessage';
+import MessageTime from '../MessageTime';
+import {ContextMenuEntry, showContextMenu} from '../../../ui/ContextMenu';
+import Icon from 'Components/Icon';
+import {t} from 'Util/LocalizerUtil';
+import {StatusType} from '../../../message/StatusType';
+import {includesOnlyEmojis} from 'Util/EmojiUtil';
+import {MessageActions} from '../MessageWrapper';
+import {Message} from 'src/script/entity/message/Message';
+
+export interface TextMessageProps extends Omit<MessageActions, 'onClickResetSession'> {
+  contextMenu: {entries: ko.Subscribable<ContextMenuEntry[]>};
+  conversation: Conversation;
+  findMessage: (conversation: Conversation, messageId: string) => Promise<ContentMessage | undefined>;
+  focusMessage?: () => void;
+  hasMarker?: boolean;
+  isLastDeliveredMessage: boolean;
+  message: ContentMessage;
+  onClickButton?: (message: ContentMessage, assetId: string) => void;
+  previousMessage: Message;
+  quotedMessage?: ContentMessage;
+  selfId: QualifiedId;
+}
+
+const ContentAsset = ({
```

Comment:
this file feels a little big, wondering how we'd feel about moving ContentAsset to its own file?